src/Changes.php

```diff
@@ -33,6 +37,40 @@ public static function empty() : self
         return $empty;
     }
 
+    /** @param Change[] $changes */
+    public static function fromIterator(iterable $changes) : self
+    {
+        $instance = new self();
+
+        $instance->changes   = [];
+        $instance->generator = (function () use ($changes) : Generator {
```

Comment:
I don't know what this is, but if some dark magic, then it is confusing and just looks like a typo, so -1. :/

Reply:
I just smashed it together while on the plane - the idea is to make the iteration repeatable.

Reply:
I was talking about ``ffunction` here:
![](https://user-images.githubusercontent.com/144181/44960712-6f7d1b00-af04-11e8-9853-92c09bf5fccc.png)
But it seems to be a rendering bug, lol.

Reply:
It is a rendering bug, yep

Reply:
This has been massively simplified now :+1: